tests/unit/v1/async/test_async_query.py

```diff
@@ -0,0 +1,380 @@
+# Copyright 2020 Google LLC All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+import types
+import aiounittest
+
+import mock
+
+from tests.unit.v1.test_base_query import _make_credentials, _make_query_response
+
+
+class TestAsyncQuery(aiounittest.AsyncTestCase):
+    @staticmethod
+    def _get_target_class():
+        from google.cloud.firestore_v1.async_query import AsyncQuery
+
+        return AsyncQuery
+
+    def _make_one(self, *args, **kwargs):
+        klass = self._get_target_class()
+        return klass(*args, **kwargs)
+
+    def test_constructor_defaults(self):
+        query = self._make_one(mock.sentinel.parent)
+        self.assertIs(query._parent, mock.sentinel.parent)
+        self.assertIsNone(query._projection)
+        self.assertEqual(query._field_filters, ())
+        self.assertEqual(query._orders, ())
+        self.assertIsNone(query._limit)
+        self.assertIsNone(query._offset)
+        self.assertIsNone(query._start_at)
+        self.assertIsNone(query._end_at)
+        self.assertFalse(query._all_descendants)
+
+    @pytest.mark.asyncio
+    async def test_get_simple(self):
```

Comment:
What does `simple` mean in this test name? I don't know that it's the responsibility of this test to verify the code that is responsible for translating a query to the protobuf request. So I'm not sure I understand what simple is alluding to.

Reply:
Similar to `test_stream_simple`, its referring to a simple usecase of `get`. I think this is confusing because there are no other `get` test cases, as the `get` implementation wraps `stream`. Maybe it would be better if `get` was tested with mocks and we scrapped this test altogether?

Reply:
I think that's probably fair since `get` just calls `stream`.

Reply:
Sounds good! I'll make that change.

Reply:
https://github.com/googleapis/python-firestore/issues/101